app/models/manageiq/providers/redfish/inventory/collector/physical_infra_manager.rb

```diff
@@ -3,5 +3,13 @@ class Inventory::Collector::PhysicalInfraManager < Inventory::Collector
     def physical_servers
       rf_client.Systems.Members
     end
+
+    def physical_racks
+      rf_client.Chassis.Members.select { |c| c.ChassisType == "Rack" }
+    end
+
+    def physical_chassis
+      rf_client.Chassis.Members.reject { |c| c.ChassisType == "Rack" }
```

Comment:
Looks like this could be cached so you don't have to make the same call twice.
Could you add a private method `chassis_members` where you `@chassis_members ||= rf_client.Chassis.Members`?
Then you can just `chassis_members.select` / `chassis_members.reject`

Reply:
Redfish client already does caching on its own, so adding another layer of cache is not strictly necessary since data is already only fetched once. But I can add it if you feel that this would make things more obvious.

Reply:
Ah okay didn't know this